### ISSUE TITLE
Set Printing Depth and Set Printing Width not working in CoqIDE

### ISSUE THREAD

**Issue by herbelin** [coq/coq#11710]
#### Description of the problem

For the record, the options `Set Printing Depth` and `Set Printing Width` do now have an effect in CoqIDE. Or, rather, they have an effect in what `coqide -debug` reports but not about what is displayed in the Messages window.

Is there something special with the Message window? Isn't its formatter the `std_formatter`?

More generally, are the options working in other user interfaces?

**Comment by JasonGross** [comment]
The options work fine in ProofGeneral, at least as of 8.10 or 8.11.  I haven't checked with newer versions.

**Comment by JasonGross** [comment]
Is this because it's an option that needs to be set via menus in CoqIDE?

**Comment by andrew-appel** [comment]
No, it is not an option that needs to be set in menus in CoqIDE, or at least, I haven't been able to find any such menu.  (I'm looking in Coq 8.11.2.)  And it's very annoying that I can't change the printing depth in the Messages window.  This is a bug!

**Comment by ppedrot** [comment]
IIRC CoqIDE adapts the printing width automatically to the size of the message window, so you don't have to do that by hand. For the printing depth, I don't know.

**Comment by ejgallego** [comment]
Indeed CoqIDE will set these options client-side, search for the calls to `Richpp.richpp_of_pp`

As @ppedrot mentioned, the current heuristics are to use the windows width, but it is very easy to use others.
